### PR TITLE
[Documentation] Documentation for trackers updated

### PR DESCRIPTION
**Pull request by mdacoca**
### Motivation and context

Documentation for AI trackers updated, new tracker added

### Checklist

- [X] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [X] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by nmanovic** [comment]
@mdacoca , need to fix issues from remark linter

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 1]
```diff
@@ -150,6 +150,18 @@ Supported DL models are not bound to the label and can be used for any objects.
...
+
+## Transformer Tracking (TransT)
+
+Simple and efficient online object tracking and segmentation tool. <br>Able to track different objects in one server request.
+Trackable object will be tracked automatically if the previous frame was a latest keyframe for the object.
+Has tracker indication on canvas. This is a modified version of the python framework PyTracking based on Pytorch.
+
+For more information, see [TransT Github](https://github.com/chenxin-dlut/TransT)
+
+![Siam Mask](/images/tracker_TransT.jpg)
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:167]:
```suggestion
![TransT](/images/tracker_TransT.jpg)
```

**Comment by nmanovic** [comment]
@mdacoca , it is not something I expect to see in the documentation.

Basically, we have 3 trackers, all of which should work more or less the same. We need to have a gif that shows how they work and describe. Try by yourself. Also need to describe each of them. Two are described. Need to describe OpenCV MIL tracker as well. Add some links which describe these trackers. For example, for MIL you can use https://learnopencv.com/tag/mil/ link. Ask me if you have more questions.

**Comment by yasakova-anastasia** [comment]
> Need to describe OpenCV MIL tracker as well. Add some links which describe these trackers. For example, for MIL you can use https://learnopencv.com/tag/mil/ link.

We have a [TrackerMIL description](https://opencv.github.io/cvat/docs/manual/advanced/opencv-tools/#trackermil) in the OpenCV Tools section. I don't think we should mix documentation in this way.

**Review by chernov-a — COMMENTED** [review]
Generally, the document looks good. Need to fix a few typos.
I like the new structure. The tabels look graphically enough to understand how the AI tools work.
I have some questions to decision about merging 2 topics (AI Tools + OpenCV Tools) into one document. As for me,  these tools even assigned on two different buttons in CVAT and it's more convinient to find the information about these tools separately.

**Comment by nmanovic** [comment]
@mdacoca , need to reduce the size of GIFs if it is possible. Documentation loading time is also a critical UX parameter. Please ask @chernov-a, he has some knowledge of how we did it previously.

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 2]
```diff
@@ -1,155 +1,309 @@
...
+Use interactors, to create a polygon semi-automatically.
 
-  ![](/images/image224.jpg)
+Supported DL models are not bound to the label, you can use them to track any object.
+
+When creating a polygon, you can use positive (regular) points
+or negative points (for some models):
+
+- Positive points are the points related to the object.
+- Negative points should be placed outside the boundary of the object.
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:45]:
```suggestion
- Negative points should be placed outside the object.
```

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 3]
```diff
@@ -1,155 +1,309 @@
...
-  ![](/images/image224.jpg)
+Supported DL models are not bound to the label, you can use them to track any object.
+
+When creating a polygon, you can use positive (regular) points
+or negative points (for some models):
+
+- Positive points are the points related to the object.
+- Negative points should be placed outside the object.
+
+![](/images/image188_detrac.jpg)
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:47]: Positive points can be anywhere on the object. But you describe here how Deep Extreme Cut algorithm expects points. 

The algorithm utilizes so-called extreme points: https://arxiv.org/pdf/1711.09081.pdf
```
This paper explores the use of extreme points in an object (left-most, right-most, top,
bottom pixels) as input to obtain precise object segmentation for images and videos. We
do so by adding an extra channel to the image in the input of a convolutional neural
network (CNN), which contains a Gaussian centered in each of the extreme points. The
CNN learns to transform this information into a segmentation of an object that matches
those extreme points.
```

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 4]
```diff
@@ -1,155 +1,309 @@
...
+4. (Optional) If the model returns masks, and you need to
+   convert masks to polygon, use the **Convert masks to polygons** toggle.
+5. Click **Interact**.
+6. Use the left click to add positive points and the right click to add negative points.
+   <br>Number of points you can add depends on the model.
+7. On the top menu, click **Done** (or **Shift+N**, **N**), and move to the next frame.
 
-- You can use the `Selected opacity` slider in the `Objects sidebar` to change the opacity of the polygon.
-  You can read more in the [Objects sidebar](/docs/manual/basics/objects-sidebar/#appearance) section.
+When you place the points, the request is sent to the server immediately.
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:67]: Each algorithm has a minimum amount of points to annotate (indeed the input data for interactors can be different). Thus when the user places the required amount of points, the request will be sent automatically to the server. If we understand that the object is complex and we need more points by default, we can use CTRL to add more points in one request.

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 5]
```diff
@@ -1,155 +1,309 @@
...
+| --------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------ |
+| Deep extreme <br>cut (DEXTR)                              | This is an optimized version of the original model, <br>introduced at the end of 2017. It uses the <br>information about extreme points of an object <br>to get its mask. The mask is then converted to a polygon. <br>For now this is the fastest interactor on the CPU.                                                         | ![](/images/dextr_example.gif) |
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:107]: Probably it will be good to add a link on the article and GitHub repo for each model. Users will be able to find more information if they need it. For example, for DEXR:
GitHub: https://github.com/scaelles/DEXTR-PyTorch
Site: https://cvlsegmentation.github.io/dextr
Article: https://arxiv.org/pdf/1711.09081.pdf

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 6]
```diff
@@ -1,155 +1,309 @@
...
+| Feature backpropagating <br>refinement <br>scheme (f-BRS) | The model allows to get a mask for an <br>object using positive points (should be <br>left-clicked on the foreground), <br>and negative points (should be right-clicked <br>on the background, if necessary). <br>It is recommended to run the model on GPU, <br>if possible.                                                     | ![](/images/fbrs_example.gif)  |
+| High Resolution <br>Net (HRNet)                           | The model allows to get a mask for <br>an object using positive points (should <br>be left-clicked on the foreground), <br> and negative points(should be <br>right-clicked on the background, <br> if necessary). <br>It is recommended to run the model on GPU, <br>if possible.                                                | ![](/images/hrnet_example.gif) |
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:109]: `negative points(should`

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 7]
```diff
@@ -1,155 +1,309 @@
...
-## Detectors
+### Detectors
 
-Detectors are used to automatically annotate one frame. Supported DL models are suitable only for certain labels.
+Use detectors to automatically annotate one frame.
 
-- Before you start, click the `magic wand` on the controls sidebar and select the `Detectors` tab.
-  You need to match the labels of the DL model (left column) with the labels in your task (right column).
-  Then click `Annotate`.
+Supported DL models are suitable only for certain labels.
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:118]: Each model is trained on a dataset. In many cases it is COCO, Pascal VOC. The model supports labels that were available in the training dataset. Need to clarify that the list of labels can be viewed on a model page for each such model. Just an idea.

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 8]
```diff
@@ -1,155 +1,309 @@
...
 
-  ![](/images/image187.jpg)
+To annotate with detectors, do the following:
 
-- Some of models supports attributes annotation (like facial emotions, for example: ``serverless/openvino/omz/intel/face-detection-0205``).
-  In this case you can also match attributes of the DL model with the attributes of a CVAT label.
+1. Click **Magic wand** ![Magic wand](/images/image189.jpg), and go to the **Detectors** tab.
+2. From the **Model** drop-down, select model (see [Detectors Models](#detectors-models)).
+3. From the left drop-down select the DL model label and match it to the right drop-down - the label of your task.
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:124]: I will recommend to add an example. For example, a DL model can have 'car' label, but our task or project can have `vehicle` label. Thus, we need to give a hint to our DL model (detector) what we want to annotate cars on images.

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 9]
```diff
@@ -1,155 +1,309 @@
...
 
-### Mask RCNN
+Some of the models support attribute annotation like facial emotions,
+for example **Attributed face detection** (`serverless/openvino/omz/intel/face-detection-0205`).
 
-The model generates polygons for each instance of an object in the image.
+In this case, you can also match the attributes of the DL model with the attributes of a CVAT label.
 
-### Faster RCNN
+![](/images/image187_1.jpg)
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:137]: The image doesn't show actual attributes. Probably need to add corresponding attributes which are supported by the model and update the screenshot.

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 10]
```diff
@@ -1,155 +1,309 @@
...
-Trackers are used to automatically annotate an object using bounding box.
-Supported DL models are not bound to the label and can be used for any objects.
+<!--lint disable maximum-line-length-->
 
-- Before you start, select the `magic wand` on the controls sidebar and go to the `Trackers` tab.
-  Then select a `Label` and `Tracker` for the object and click `Track`. Then annotate the desired objects with the
-  bounding box in the first frame.
+| Model       | Description                                                                                                                                               |
+| ----------- | --------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Mask RCNN   | The model generates polygons for each instance of an object in the image.                                                                                 |
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:149]: We definitely supports more detectors. See the full list here: https://github.com/opencv/cvat#deep-learning-serverless-functions-for-automatic-labeling

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 11]
```diff
@@ -1,155 +1,309 @@
...
+| SiamMask                      | Fast online Object Tracking and Segmentation.<br>Tracker can track different objects in one server request. <br>Trackable object will be tracked automatically <br>if the previous frame was the latest keyframe for the object. <br>Has a tracker indication on canvas. <br>SiamMask tracker supported CUDA.                                                                                                                                                      |
+| Transformer Tracking (TransT) | Simple and efficient online object tracking and segmentation tool. <br>Able to track different objects in one server request. <br>Trackable object will be tracked automatically if the previous<br> frame was the latest keyframe for the object. <br>Has tracker indication on canvas. <br>This is a modified version of the python framework PyTracking based on Pytorch. <br>For more information, see [TransT Github](https://github.com/chenxin-dlut/TransT) |
+
+<!--lint enable maximum-line-length-->
+
+> If you plan to track simple non-overlapping objects consider using fast client-side [TrackerMIL from OpenCV](#trackermil).
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:203]: Need to add MIL tracker into the table above with the appropriate description.

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 12]
```diff
@@ -1,155 +1,309 @@
...
+2. Click on **Histogram equalization** button.
+   <br>![](/images/image221.jpg)
+
+Then contrast of the current frame will be improved.
+If you change the frame, it will be equalized too.
+You can disable equalization by clicking the **Histogram equalization** button again.
+
+![](/images/image222.jpg)
+
+### TrackerMIL
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:278]: I don't understand why we have here information about MIL tracker. It should be in the tracker section only.

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 13]
```diff
@@ -1,155 +1,309 @@
...
+  by temporarily disabling the automatic line creation. To do that, click `Ctrl`.
+
+- You can change the number of points in the polygon with slider:
+
+  ![](/images/image224.jpg)
+
+- To change polygon opacity, on the [Objects sidebar](/docs/manual/basics/objects-sidebar/#appearance)
+  use the **Selected opacity** slider.
+
+### Histogram Equalization
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:260]: I will suggest creating a separate page and description of how we can adjust our original frame. Histogram Equalization is about that.

image

Comment (nmanovic) [reply site/content/en/docs/manual/advanced/ai-tools.md:260]: On the same page we can description the image grid, brightness, contract, and saturation.

File: site/content/en/docs/manual/advanced/ai-tools.md [hunk 14]
```diff
@@ -1,155 +1,309 @@
...
+To start working with OpenCV, you need to load the OpenCV library first.
+<br>To do this, on the [controls sidebar](/docs/manual/basics/controls-sidebar/),
+click **OpenCV**![OpenCV](/images/image201.jpg)
+<br>Library will be downloaded automatically.
+
+![](/images/image198.jpg)
+
+Once it is loaded, the tool's functionality will be available.
+
+### Intelligent scissors
```
Comment (nmanovic) [inline site/content/en/docs/manual/advanced/ai-tools.md:220]: I'm not sure, but probably it should be a part of the section with interactors.

**Review by nmanovic — CHANGES_REQUESTED** [review]
I like the structure. Need to improve the content.

File: site/content/en/docs/manual/advanced/ai-tools.md (hunk 13 above)
Comment (mdacoca) [reply site/content/en/docs/manual/advanced/ai-tools.md]: I agree, but suggest making different task/request for this one

**Review by nmanovic — APPROVED** [review]
